### PR TITLE
Feat/eth savings

### PR DESCRIPTION
**Pull request by jmichel-bc**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: config/mocks/wallet-options-v4.json [hunk 1]
```diff
@@ -28,6 +28,24 @@
...
+        },
+        "simplebuy": {
+          "eth": "0xFF202CBC6C794516164E45621143acDe4a3FEa24"
+        },
+        "lending": {
+          "eth": "0xBA0145434A0a53E58D10680254756F679c359463"
+        },
+        "rewards": {
+          "eth": "0x7d9d70c1A99FcC952d48b781947FAa0a3795083E"
+        }
```
Comment (plondon) [inline config/mocks/wallet-options-v4.json:48]: cc @schnogz should we also set up staging.js, development.js, and production.js values and apply in webpackBuild?

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 2]
```diff
@@ -434,8 +435,20 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
           throw new Error('Missing deposit address')
         }
 
+        const hotWalletAddress = selectors.core.walletOptions
+          .getHotWalletAddresses(yield select(), Product.REWARDS)
+          .getOrFail('No extra data') as String
+        if (typeof hotWalletAddress !== 'string') {
+          throw new Error('Hot Wallet Address not located.')
```
Comment (mpfluger) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:442]: Old way of sending in the `if` and log, new way with replaced address and memo in the `else` branch

File: packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts [hunk 3]
```diff
@@ -196,7 +196,19 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas; network
       const payment = paymentGetOrElse(BASE.coin, paymentR)
       if (onChain) {
         try {
-          yield call(buildAndPublishPayment, payment.coin, payment, order.kind.depositAddress)
+          const hotWalletAddress = selectors.core.walletOptions
+            .getHotWalletAddresses(yield select(), Product.SWAP)
+            .getOrFail('No extra data') as String
+          if (typeof hotWalletAddress !== 'string') {
```
Comment (mpfluger) [inline packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts:202]: Same as above

**Review by mpfluger — APPROVED** [review]

File: config/mocks/wallet-options-v4.json (hunk 1 above)
Comment (schnogz) [reply config/mocks/wallet-options-v4.json:48]: i think this is covered via the applications PR that was created and merged

**Review by schnogz — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts [hunk 4]
```diff
@@ -49,6 +50,10 @@ export default ({ coreSagas, networks }: { coreSagas: any; networks: any }) => {
         updatedPayment = yield updatedPayment.memoType('text')
         // @ts-ignore
         updatedPayment = yield updatedPayment.setDestinationAccountExists(true)
+      } else if (hotwalletAddress && payment.coin === 'ETH') {
```
Comment (paulo-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts:53]: Will this also be used for ERC20 coins?

Comment (jmichel-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.utils.ts:53]: Yes, in the case of ERC20's this payment.coin value is also 'ETH'.

**Pull request merged**